### PR TITLE
[DEV-10842] Implement new contact card layouts

### PR DESCRIPTION
**Pull request by kudlajz**
This PR implements the required design changes to support both layouts and optionally visible avatar. I was unable to reuse the content renderer due to this theme having a slightly different card layout.

Screenshot 2023-05-23 at 20 13 50

### CODE REVIEW

File: components/SlateRenderer/SlateRenderer.module.scss [hunk 1]
```diff
@@ -11,8 +11,10 @@
         .prezly-slate-document {
             color: inherit;
 
-            > *:last-child {
-                margin-bottom: 0 !important;
+            @include tablet-up {
```
Comment (oleg-semyonov) [inline components/SlateRenderer/SlateRenderer.module.scss:14]: Why was this added?

If there's not enough space on the bottom, it should be changed with padding in the Layout or Story component

File: components/ContactCard/ContactCard.module.scss [hunk 2]
```diff
@@ -7,45 +7,63 @@
...
+    gap: $spacing-3;
     flex-wrap: wrap;
 
     &:not(:last-child) {
-        margin-bottom: $spacing-2;
+        margin-bottom: $spacing-4;
+    }
+
+    &.social {
+        gap: $spacing-3;
```
Comment (oleg-semyonov) [inline components/ContactCard/ContactCard.module.scss:47]: This duplicates the rule on line 39

File: components/ContactCard/ContactCard.tsx [hunk 3]
```diff
@@ -13,67 +12,96 @@ import styles from './ContactCard.module.scss';
...
                     )}
-                </div>
-                <div className={styles.linkGroup}>
                     {facebook && (
                         <a
                             href={`https://facebook.com/${facebook}`}
                             className={classNames(styles.link)}
                         >
                             <IconFacebook width={16} height={16} className={styles.icon} />
-                            <span className={styles.linkText}>{facebook}</span>
```
Comment (oleg-semyonov) [inline components/ContactCard/ContactCard.tsx:76]: Was removing the social text a design decision?
In either way, it would be nice to add a `title` to the links to mace it more accessible

File: components/ContactCard/ContactCard.module.scss [hunk 4]
```diff
@@ -7,45 +7,63 @@
...
+
+        .link {
+            color: $color-base-600;
+        }
+    }
+
+    .signature &:not(.social) {
```
Comment (oleg-semyonov) [inline components/ContactCard/ContactCard.module.scss:54]: This selector looks incorrect. It would compile to `.linkGroup .signature`, but I guess you wanted `.linkGroup.signature`. Also, the `&:not` selector would apply to children of `.signature` class, not itself

```suggestion
    &.signature:not(.social) {
```

File: components/ContactCard/ContactCard.module.scss [hunk 5]
```diff
@@ -59,18 +77,28 @@
...
     height: 1rem;
     flex-shrink: 0;
-    margin-right: $spacing-1;
 }
 
 .container {
-    @include border-radius-lg;
-
-    padding: $spacing-4;
     background: $color-base-50;
```
Comment (oleg-semyonov) [inline components/ContactCard/ContactCard.module.scss:83]: I guess the background should also be omitted for signature layout?

File: components/ContactCard/ContactCard.tsx [hunk 6]
```diff
@@ -13,67 +12,96 @@ import styles from './ContactCard.module.scss';
...
-                <div className={styles.linkGroup}>
-                    {email && (
-                        <a href={`mailto:${email}`} className={styles.link}>
-                            <IconEmail className={styles.icon} />
-                            <span className={styles.linkText}>{email}</span>
-                        </a>
-                    )}
-                    {website && (
-                        <a href={website} className={styles.link}>
+                {isCard && (
```
Comment (oleg-semyonov) [inline components/ContactCard/ContactCard.tsx:47]: I don't quite get why the links are different for two layouts 🤔

**Review by oleg-semyonov — COMMENTED** [review]
I might miss some context, but many of these change look weird to me 🤔

File: components/ContactCard/ContactCard.tsx (hunk 6 above)
Comment (kudlajz) [reply components/ContactCard/ContactCard.tsx:47]: That's a question for @iamhugofelix  I guess 🙂

File: components/ContactCard/ContactCard.module.scss (hunk 4 above)
Comment (kudlajz) [reply components/ContactCard/ContactCard.module.scss:54]: It's correct or at least it works the way I want it to. What I wanted is this:
```
.signature .linkGroup:not(.social)
```

In other words - if the parent has `.signature` class (in my case `.container`), apply these rules to all `.linkGroup` elements except if it has `.linkGroup.social`.

File: components/ContactCard/ContactCard.module.scss (hunk 5 above)
Comment (kudlajz) [reply components/ContactCard/ContactCard.module.scss:83]: Possibly but I haven't noticed any difference by removing this rule 🤔

File: components/ContactCard/ContactCard.tsx (hunk 3 above)
Comment (kudlajz) [reply components/ContactCard/ContactCard.tsx:76]: Yes, that's how it was designed. Not sure why the handles were omitted in this theme since we have them everywhere else in the new designs. Maybe we should keep them @iamhugofelix?

File: components/ContactCard/ContactCard.module.scss (hunk 4 above)
Comment (oleg-semyonov) [reply components/ContactCard/ContactCard.module.scss:54]: Oh, I got a bit confused, was thinking this is applied to the container class. Sorry!

File: components/SlateRenderer/SlateRenderer.module.scss (hunk 1 above)
Comment (kudlajz) [reply components/SlateRenderer/SlateRenderer.module.scss:14]: It's because I've noticed it looks bad on mobile:

Screenshot 2023-05-24 at 9 31 32

On desktop, there's enough space at the bottom due to padding but on mobile, [that padding is reset to 0](https://github.com/prezly/theme-nextjs-lena/blob/main/modules/Layout/Layout.module.scss#L21).

So I guess a better solution would be to keep the bottom padding? Something like 2rem looks like this:
Screenshot 2023-05-24 at 9 33 54

Comment (oleg-semyonov) [reply components/SlateRenderer/SlateRenderer.module.scss:14]: Yeah, that works well! No idea why the padding was reset for mobile 🤷🏻

File: components/ContactCard/ContactCard.tsx (hunk 3 above)
Comment (kudlajz) [reply components/ContactCard/ContactCard.tsx:76]: I've added the titles for now, at least.

File: components/ContactCard/utils.ts [hunk 7]
```diff
@@ -18,3 +18,11 @@ export function getSocialHandles(contactInfo: ContactInfo) {
         twitter: isValidTwitter ? socialLinks.getProfileId('twitter', twitter) : null,
     };
 }
+
+export function getUrl(url: string): URL | null {
```
Comment (oleg-semyonov) [inline components/ContactCard/utils.ts:22]: I remember adding URL normalization to content renderer for URLs without protocol. It should be applied as a transform brefore the render step, if I remember correctly. Has it been broken? 🤔

Comment (kudlajz) [reply components/ContactCard/utils.ts:22]: I am not sure but I've noticed that the URL object is not being created correctly for some Prezly newsrooms so I've used the try/catch in the content renderer and migrated it here just in case.

File: components/ContactCard/ContactCard.tsx (hunk 3 above)
Comment (iamhugofelix) [reply components/ContactCard/ContactCard.tsx:76]: Ok, here's the issue. I've followed the [design in Figma](https://www.figma.com/file/qGvtTUikFEzusn34lnaspD/Theme%3A-Lena?type=design&node-id=59%3A1460&t=L5zEg8USsoyTD0L9-1) (where the handles are not shown. As I already told @kudlajz I wanted to make as less changes to the cards as possible and that's the reason for not "making any changes".

TBH it is the same if we have the handles or not, it doesn't change it that much, so feel free to add them.

File: components/ContactCard/ContactCard.tsx (hunk 6 above)
Comment (iamhugofelix) [reply components/ContactCard/ContactCard.tsx:47]: I'm not sure I got what you meant?
